### PR TITLE
IBX-5663: Paginated role assignments to improve performance in the Roles module

### PR DESCRIPTION
**Pull request by barw4**
| Question      | Answer
| ------------- | ---
| Tickets       | https://issues.ibexa.co/browse/IBX-5663
| Improvement?      | yes
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

The new changes utilize some API methods introduced in the linked PR, like `countRoleAssignments` and  `loadRoleAssignments`. Using pagination should help with the performance in the Role module.

Required PR: https://github.com/ezsystems/ezplatform-kernel/pull/373

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

**Review by konradoboza — APPROVED** [review]
Is failing CI job random, or does it require some further investigation?

File: src/bundle/Controller/RoleController.php [hunk 1]
```diff
@@ -132,14 +121,14 @@ public function viewAction(Request $request, Role $role, int $policyPage = 1, in
...
+            $assignmentsCount = $this->roleService->countRoleAssignments($role);
         } catch (UnauthorizedException $e) {
-            $assignments = [];
+            $assignmentsCount = 0;
         }
 
         return $this->render('@ezdesign/user/role/index.html.twig', [
             'role' => $role,
-            'assignments' => $assignments,
+            'assignmentsCount' => $assignmentsCount,
```
Comment (dew326) [inline src/bundle/Controller/RoleController.php]:
```suggestion
            'assignments_count' => $assignmentsCount,
```

File: src/bundle/Resources/views/themes/admin/user/role/index.html.twig [hunk 2]
```diff
@@ -32,7 +32,7 @@
                         </li>
                         <li role="presentation" class="nav-item">
                             <a href="#users-and-groups" class="nav-link" role="tab" data-toggle="tab">
-                                {{ 'role_assignment.view.list.title.count'|trans({'%count%': assignments|length})|desc('Assignments (%count%)') }}
+                                {{ 'role_assignment.view.list.title.count'|trans({'%count%': assignmentsCount})|desc('Assignments (%count%)') }}
```
Comment (dew326) [inline src/bundle/Resources/views/themes/admin/user/role/index.html.twig]:
```suggestion
                                {{ 'role_assignment.view.list.title.count'|trans({'%count%': assignments_count})|desc('Assignments (%count%)') }}
```

File: src/bundle/Resources/views/themes/admin/user/role/index.html.twig [hunk 3]
```diff
@@ -51,7 +51,8 @@
                     {{ render(controller('EzSystems\\EzPlatformAdminUiBundle\\Controller\\RoleAssignmentController::listAction', {
                         roleId: role.id,
                         assignmentPage: assignment_page,
-                        routeName: route_name
+                        routeName: route_name,
+                        assignmentsCount: assignmentsCount
```
Comment (dew326) [inline src/bundle/Resources/views/themes/admin/user/role/index.html.twig]:
```suggestion
                        assignmentsCount: assignments_count
```

**Comment by barw4** [comment]
> Is failing CI job random, or does it require some further investigation?

It requires related kernel PR: https://github.com/ezsystems/ezplatform-admin-ui/actions/runs/5144411939/jobs/9260679669?pr=2102#step:16:740

**Review by dew326 — APPROVED** [review]

**Review by micszo — APPROVED** [review]

**Pull request merged**